### PR TITLE
Add a test for WithAzureRegion(null)

### PR DESCRIPTION
**Pull request by bgavrilMS**
Just a unit test

### CODE REVIEW

**Review by SameerK-MSFT — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/ClientCredentialWithRegionTests.cs [hunk 1]
```diff
@@ -236,6 +236,25 @@ public async Task RegionFallbackToGlobalAsync()
             }
         }
 
+        [TestMethod]
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit/PublicApiTests/ClientCredentialWithRegionTests.cs:239]: nit: you could do ` [TestMethod] [DataRow()]` to have less code. and then it's clear what's being covered.

**Review by jennyf19 — APPROVED** [review]

**Review by trwalke — APPROVED** [review]
LGTM
